### PR TITLE
Ignore k8s node updates

### PR DESCRIPTION
**Pull request by aanm**
```release-note
Ignore unnecessary Kubernetes node updates
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5170)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

File: pkg/node/node.go [hunk 1]
```diff
@@ -249,3 +249,48 @@ func (n *Node) OnDelete() {
 func (n *Node) IsLocal() bool {
 	return n != nil && n.Name == GetName()
 }
+
+// PublicAttrEquals returns true only if the public attributes of both nodes
+// are the same otherwise returns false.
+func (n *Node) PublicAttrEquals(o *Node) bool {
+	if n == nil {
```
Comment (ianvernon) [inline pkg/node/node.go]: Please add unit tests for this.

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

File: pkg/node/node.go (hunk 1 above)
Comment (aanm) [reply pkg/node/node.go]: Done

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**